smarts/core/traffic_history_provider.py

```diff
@@ -69,10 +73,12 @@ def step(self, provider_actions, dt, elapsed_sim_time) -> ProviderState:
         if (
             not self._current_traffic_history
             or timestamp is None
-            or str(timestamp) not in self._current_traffic_history
+            or str(round(timestamp + self.start_time_offset, 1))
```

Comment:
I would suggest moving `time_with_offset` before this if statement and using it in place of the rounded value here.

Reply:
the timestamp `or timestamp is None` check here might be `None` which causes an error when adding None with self.start_time_offset

Reply:
I guess I could move this check `str(round(timestamp + self.start_time_offset, 1)) not in self._current_traffic_history` to be after `time_with_offset` definition?

Reply:
Oh sure, I missed that.